tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java

```diff
@@ -324,6 +332,46 @@ private void setBreakpoints(List<BallerinaTestDebugPoint> breakPoints) throws Ba
         }
     }
 
+    protected Variable[] fetchDebugHitVariables(StoppedEventArguments args) throws BallerinaTestException {
+        if (!DebugHitListener.connector.isConnected()) {
+            return null;
```

Comment:
Shall we either return an empty `Variable[]` array or raise a `BallerinaTestException` instead, to avoid the potential NPE?

Reply:
Fixed by returning an empty Variable[].